### PR TITLE
feat : add post, delete post, edit post api

### PR DESCRIPTION
**Pull request by mikesch803**
1. user can add post
2. user can edit post
3. user can delete post

### CODE REVIEW

File: src/pages/home/postSlice.js [hunk 1]
```diff
@@ -16,9 +19,91 @@ export const getAllPosts = createAsyncThunk("posts/getAllPosts", async () => {
...
+    } catch (err) {
+      return rejectWithValue(err);
+    }
+  }
+);
+
+export const editPost = createAsyncThunk(
+  "posts/editPost",
+  async (post, { rejectWithValue }) => {
+    console.log(post)
```
Comment (ApurvaSawant11) [inline src/pages/home/postSlice.js:72]: You can remove this.

**Review by ApurvaSawant11 — APPROVED** [review]
Looks good to me! You can merge the PR

File: src/pages/home/Home.jsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 import { useDispatch, useSelector } from "react-redux";
 import { Aside, Loader, Nav, PostCard, Textbox } from "../../components";
+import { EditPost } from "../../components/edit-textbox/EditPost";
 import "./Home.css";
-import { getAllPosts } from "./postSlice";
+import { getAllPosts, closeEditModal } from "./postSlice";
 export function Home() {
-  const { posts, loading } = useSelector((state) => state.posts);
+  const { posts, loading, editModal, currPost } = useSelector((state) => state.posts);
+console.log(editModal)
```
Comment (foolhardy21) [inline src/pages/home/Home.jsx:9]: `console.log`

**Review by foolhardy21 — APPROVED** [review]
Use of `async thunk` looks good along with rest of the code. Good to go ahead.

**Pull request merged**